src/modules/claims/claims.service.ts

```diff
@@ -757,7 +757,20 @@ export class ClaimsService {
     registrationTypes = [RegistrationTypes.OffChain],
     claim,
   }: PublishPublicClaimOptions): Promise<string | undefined> {
-    claim.token = claim.token || token;
+    claim.token = claim.token || (token as string);
+    const payload = (await this._didRegistry.decodeJWTToken({
```

Comment:
It make sense to decode and validate token in separate method, because such operation is performed in several places

Reply:
I would make it part of `DidRegistry.decodeJwtToken`

Reply:
I'm not sure what you mean about validate token. It is already decoding in that decodeJwtToken helper.

Reply:
By validation I mean checking that payload has `iss` and `claimData`

Reply:
@JGiter and are you saying you want to throw an error if those are not found?

Reply:
Yes, `publishPublicClaim` expect `iss` and `claimData` in `payload`

Reply:
> I would make it part of `DidRegistry.decodeJwtToken`

This method is used for on and off chain claims, and a claim that is on-chain registration only will not have a token to decode. The test for 'should be able to issue and publish onchain without request' would not pass as an error would be thrown that there is no token to validate. I put this in the off-chain logic only, and made a separate validate method for the decoded token.